src/Repository/TagRepository.php

```diff
@@ -85,6 +88,36 @@ public function findAllFlatTagsWithNbEntries($userId)
             ->getArrayResult();
     }
 
+    /**
+     * Find all tags per user with nb entries.
+     *
+     * @param int $userId
+     *
+     * @return array<array{tag: Tag, nbEntries: int}>
+     */
+    public function findAllTagsWithNbEntries($userId)
+    {
+        $rsm = new ResultSetMappingBuilder($this->getEntityManager());
+        $rsm->addRootEntityFromClassMetadata(Tag::class, 't');
+        $rsm->addEntityResult(Tag::class, 't', 'tag');
+        $rsm->addScalarResult('nbEntries', 'nbEntries', 'integer');
+
+        $sql = <<<SQL
+            SELECT DISTINCT {$rsm->generateSelectClause()}, count(e.id) as nbEntries
+            FROM {$this->tablePrefix}tag t
+            LEFT JOIN {$this->tablePrefix}entry_tag et ON et.tag_id = t.id
+            JOIN {$this->tablePrefix}entry e ON e.id = et.entry_id
+            WHERE e.user_id = :userId
+            GROUP BY t.id
+            ORDER BY t.label
+        SQL;
```

Comment:
Just to be sure, this is working properly on each database?

Reply:
I'd say yes, as it's used only by the `/tag/list` route, that is called multiple times in TagControllerTest, and CI proves it runs properly on each DBMS